lib/roo_on_rails/routemaster/publisher.rb

```diff
@@ -0,0 +1,49 @@
+module RooOnRails
+  module Routemaster
+    class Publisher
+      attr_reader :model, :event
+
+      def initialize(model, event)
+        @model = model
+        @event = event
+      end
+
+      def publish?
+        true
+      end
+
+      def publish!
+        return unless publish?
+
+        event_data = data # cache in case it isn't memoised
```

Comment:
erm, "cache" ? "memoise"? `data` is a stub/virtual method so this doesn't actually do anything. I think you could replace lines 18-21 with just:

```ruby
routemaster.send(event, topic, url, data: recustive_stringify(event_data))
```